### PR TITLE
New Party Loading Messages

### LINKED ISSUE #7825: New Party Loading Text
It would be fun to have the text that displays when you are loading a party be more flavorful.

Possible text:

"Your party is being summoned. Please wait..."

Open to other ideas! Maybe we could even have a couple different ones?

### PR DESCRIPTION
**Pull request by khipkin**
Fixes #7825

_(edit: problem solved)_

Adds alternate strings to display while the user's party is loading:

image
image
image
image

### CODE REVIEW

File: website/views/options/social/party/loading-new-party.jade [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .container-fluid
   .row.text-center: .col-xs-12
-    h2=env.t('loadingNewParty')
+    h2=env.t('partyLoading' + partyLoadingMessage)
```
Comment (crookedneighbor) [inline website/views/options/social/party/loading-new-party.jade]: This is referencing a server variable (hence undefined) instead of a client variable.

``` jade
h2 {{::window.env.t('partyLoading' + partyLoadingMessage)}}
```

**Comment by ghost** [comment]
I may be overcautious, but I'd like to ask Socialites about the following messages:
- Your party is searching for scrolls. Please wait...
- Your party is counting their loot. Please wait...

am I the only one who has a feeling that a user (a newbie) may expect a chance of finding / getting something when the party is finally loaded - and begin asking questions why they never find scrolls.

Maybe rephrase it like:
- Your party is counting for scrolls. Please wait...
- Your party is hiding their loot. Please wait...

**Comment by Alys** [comment]
I think @GitHubSphinx is correct. We've found over the years that some users do get confused by gamified text and those two items are likely to prompt questions. I think even the "counting for scrolls" and "hiding their loot" versions might be confusing if the user thinks they're meant to be meaningful. I'm sorry, @khipkin, I should have commented on that in the issue.

**Comment by ghost** [comment]
For what it's worth, I've meant "counting the scrolls". Sorry for the typo

File: website/views/options/social/party/loading-new-party.jade (hunk 1 above)
Comment (khipkin) [inline website/views/options/social/party/loading-new-party.jade]: That didn't quite work, but it sent me in the right direction! Thanks, I didn't know client- vs server-side variables were a thing.

**Comment by khipkin** [comment]
@Alys @GitHubSphinx Those concerns make a lot of sense.  I'll remove those troublesome strings!

**Comment by khipkin** [comment]
image
image
image
image

**Comment by khipkin** [comment]
Unless there are further desired changes to the messages, this is done.

File: common/locales/en/partyLoadingMessages.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (crookedneighbor) [inline common/locales/en/partyLoadingMessages.json]: Adding new files to Transifex is kind of a pain. I think these can just live in the groups json file for now.

File: website/client/js/controllers/partyCtrl.js [hunk 3]
```diff
@@ -3,15 +3,20 @@
...
       $scope.type = 'party';
       $scope.text = window.env.t('party');
-      $scope.group = {loadingParty: true}
+      $scope.group = {loadingParty: true};
 
       $scope.inviteOrStartParty = Groups.inviteOrStartParty;
       $scope.loadWidgets = Social.loadWidgets;
 
+      // Random message between 1 and PARTY_LOADING_MESSAGES
+      $scope.partyLoadingMessage = Math.floor(Math.random() * PARTY_LOADING_MESSAGES) + 1;
```
Comment (crookedneighbor) [inline website/client/js/controllers/partyCtrl.js]: Nit pick, but this is returning a number, not the the actual message. How about:

``` js
var partyMessageNumber = Math.floor(Math.random() * PARTY_LOADING_MESSAGES) + 1;
$scope.partyLoadingMessage = window.env.t('partyLoading' + partyMessageNumber);
```

Then you can just do

``` jade
h2 {{::partyLoadingMessage}}
```

Comment (khipkin) [inline website/client/js/controllers/partyCtrl.js]: Agreed.  But I was going off of the example set by the [loading-screen tip code](https://github.com/HabitRPG/habitrpg/blob/29498976ee2e38a797f0ab0482b220a22c0b3008/website/server/controllers/top-level/pages.js#L27).

**Comment by khipkin** [comment]
Updated based on feedback.

**Comment by mandyzhou** [comment]
This is super nitpicky and not at all important, so feel free to ignore, but "Your party is coming in from battle" sounds a little awkward to me. How about "Your party is returning from battle"?

File: common/locales/en/groups.json [hunk 4]
```diff
@@ -37,7 +37,10 @@
     "invitedToNewParty": "You were invited to join a party! Do you want to leave this party and join <%= partyName %>?",
     "joinNewParty": "Join New Party",
     "declineInvitation": "Decline Invitation",
-    "loadingNewParty": "Your party is loading. Please wait...",
+    "partyLoading1": "Your party is being summoned. Please wait...",
+    "partyLoading2": "Your party is coming in from battle. Please wait...",
+    "partyLoading3": "Your party is gathering. Please wait...",
+    "partyLoading4": "Your party is materializing. Please wait...",
```
Comment (crookedneighbor) [inline common/locales/en/groups.json:43]: As soon as we get the OK from @Alys and @lemoness on the wording for these, it can be merged.

Comment (Alys) [inline common/locales/en/groups.json:43]: I think these are good to go. We'll watch for comments about them and see how they go down.

**Comment by crookedneighbor** [comment]
Notated toward your next tier!
